packages/connection/src/Manager.php

```diff
@@ -28,6 +29,155 @@ class Manager implements Manager_Interface {
+				add_action( 'login_form_jetpack_json_api_authorization', array( &$this, 'login_form_json_api_authorization' ) );
+				add_filter( 'xmlrpc_methods', array( $this, 'public_xmlrpc_methods' ) );
+			} else {
+				add_action( 'rest_api_init', array( $this, 'initialize_rest_api_registration_connector' ) );
+			}
+		}
+	}
+
+	/**
+	 * Sets up the XMLRPC request handlers.
+	 *
+	 * @param Array                  $request_params incoming request parameters.
+	 * @param Boolean                $is_active whether the connection is currently active.
+	 * @param Boolean                $is_signed whether the signature check has been successful.
+	 * @param \Jetpack_XMLRPC_Server $xmlrpc_server (optional) an instance of the server to use instead of instantiating a new one.
+	 */
+	public function setup_xmlrpc_handlers(
+		$request_params,
+		$is_active,
+		$is_signed,
+		\Jetpack_XMLRPC_Server $xmlrpc_server = null
+	) {
+		if (
+			! isset( $request_params['for'] )
+			|| 'jetpack' !== $request_params['for']
+		) {
+			return false;
+		}
+
+		// Alternate XML-RPC, via ?for=jetpack&jetpack=comms.
+		if (
+			isset( $request_params['jetpack'] )
+			&& 'comms' === $request_params['jetpack']
+		) {
+			if ( ! Constants::is_defined( 'XMLRPC_REQUEST' ) ) {
+				// Use the real constant here for WordPress' sake.
+				define( 'XMLRPC_REQUEST', true );
+			}
+
+			add_action( 'template_redirect', array( $this, 'alternate_xmlrpc' ) );
+
+			add_filter( 'xmlrpc_methods', array( $this, 'remove_non_jetpack_xmlrpc_methods' ), 1000 );
+		}
+
+		if ( ! Constants::get_constant( 'XMLRPC_REQUEST' ) ) {
+			return false;
+		}
+		// Display errors can cause the XML to be not well formed.
+		@ini_set( 'display_errors', false ); // phpcs:ignore
+
+		if ( $xmlrpc_server ) {
+			$this->xmlrpc_server = $xmlrpc_server;
+		} else {
+			require_once JETPACK__PLUGIN_DIR . 'class.jetpack-xmlrpc-server.php';
+			$this->xmlrpc_server = new Jetpack_XMLRPC_Server();
+		}
+
+		$this->require_jetpack_authentication();
+
+		if ( $is_active ) {
+			// Hack to preserve $HTTP_RAW_POST_DATA.
+			add_filter( 'xmlrpc_methods', array( $this, 'xmlrpc_methods' ) );
+
+			if ( $is_signed ) {
+				// The actual API methods.
+				add_filter( 'xmlrpc_methods', array( $this->xmlrpc_server, 'xmlrpc_methods' ) );
+			} else {
+				// The jetpack.authorize method should be available for unauthenticated users on a site with an
+				// active Jetpack connection, so that additional users can link their account.
+				add_filter( 'xmlrpc_methods', array( $this->xmlrpc_server, 'authorize_xmlrpc_methods' ) );
+			}
+		} else {
+			// The bootstrap API methods.
+			add_filter( 'xmlrpc_methods', array( $this->xmlrpc_server, 'bootstrap_xmlrpc_methods' ) );
+
+			if ( $is_signed ) {
+				// The jetpack Provision method is available for blog-token-signed requests.
+				add_filter( 'xmlrpc_methods', array( $this->xmlrpc_server, 'provision_xmlrpc_methods' ) );
+			} else {
+				new XMLRPC_Connector( $this );
```

Comment:
Silly question perhaps, but do we need to do anything with the connector object references? Seems like we don't use them anywhere and are only interested in instantiating them.

Reply:
I think we're good to keep this as-is.